src/coreclr/hosts/coreshim/CoreShim.cpp

```diff
@@ -158,39 +191,12 @@ HRESULT coreclr::GetCoreClrInstance(_Outptr_ coreclr **instance, _In_opt_z_ cons
         return S_FALSE;
     }
 
-    // Since the CoreShim is being loaded, there is a chance the scenario depends on
-    // other aspects of the offical host platform (e.g. hostpolicy). Verify a hostpolicy
-    // is _not_ already loaded and if not, attempt to load a hostpolicy library adjacent
-    // to the coreshim. If there isn't one, just keep going.
-    const WCHAR *hostpolicyName = W("hostpolicy.dll");
-    HMODULE hMod = ::GetModuleHandleW(hostpolicyName);
-    if (hMod == nullptr)
+    const wchar_t* mockHostPolicyEnvVar = W("MOCK_HOSTPOLICY");
+    if (::GetEnvironmentVariableW(mockHostPolicyEnvVar, nullptr, 0) != 0)
```

Comment:
Why the double call? This makes no sense. Please just check for an empty string.

Reply:
The double call is because the implementation of `GetEnvVar` throws if the length is `0`. I'll make that throw optional and remove the double call.